### PR TITLE
fix: bucker

### PR DESCRIPTION
**Pull request by mungomat**
We talked about this issue at Froscon. 

bucker "always" does not return a unique id in case of exact same timestamps (see pull request backup_time)

Tobias

### CODE REVIEW

File: internal/restic/snapshot_policy.go [hunk 1]
```diff
@@ -90,7 +90,9 @@ func ApplyPolicy(list Snapshots, p ExpirePolicy) (keep, remove Snapshots) {
 		{p.Yearly, y, -1},
 	}
 
+	var nr int
 	for _, cur := range list {
```
Comment (fd0) [inline internal/restic/snapshot_policy.go]: We can use the index from the list here:
```
for nr, cur := range list {
```

Then we don't need another variable.

File: internal/restic/snapshot_policy.go [hunk 2]
```diff
@@ -103,7 +105,7 @@ func ApplyPolicy(list Snapshots, p ExpirePolicy) (keep, remove Snapshots) {
 		// Now update the other buckets and see if they have some counts left.
 		for i, b := range buckets {
 			if b.Count > 0 {
-				val := b.bucker(cur.Time)
+				val := b.bucker(cur.Time, nr)
```
Comment (fd0) [inline internal/restic/snapshot_policy.go]: Good idea, I like this solution!

**Review by fd0 — CHANGES_REQUESTED** [review]
Thanks a lot for submitting this PR. I've added a few small things. The only remaining blocker is that there is no test yet for this special case. Would you mind adding a test?

Please let me know if you need any help.

File: internal/restic/snapshot_policy.go [hunk 3]
```diff
@@ -90,7 +90,8 @@ func ApplyPolicy(list Snapshots, p ExpirePolicy) (keep, remove Snapshots) {
 		{p.Yearly, y, -1},
 	}
 
-	for _, cur := range list {
+	for nr, cur := range list {
+		nr = nr + 1
```
Comment (fd0) [inline internal/restic/snapshot_policy.go]: hm, now `nr` is always incremented, so for the first element of `list`, `nr` is set to `1`. Was that intentional?

**Review by fd0 — CHANGES_REQUESTED** [review]

File: internal/restic/snapshot_policy.go (hunk 3 above)
Comment (mungomat) [reply internal/restic/snapshot_policy.go]: You are right, we dont need this line

**Comment by mungomat** [comment]
hm, the new tests fail - I think we have to update some policy_keep_snapshots-files?

**Comment by fd0** [comment]
Ah, yes. The idea is that you run the tests once with `-update=true` so the policy files are updated. Then have a look at what has changed (and if that's correct), then commit the files.

**Comment by fd0** [comment]
I've updated the files for you, thanks for your contribution, looks good now!

**Review by fd0 — APPROVED** [review]

**Comment by fd0** [comment]
Merged, thanks!

**Pull request merged**